packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1089,16 +1104,68 @@ export function getSdkModelPropertyType(
       })
     );
   }
-  return diagnostics.wrap({
+  const multiPartOption: MultipartOptionsType = {
+    isNameDefined: httpOperationPart ? httpOperationPart.name !== undefined : true,
+    isFilePart: isBytesInput && operationIsMultipart,
+    multi: httpOperationPart ? httpOperationPart.multi : isRawBytesArray,
+    headers: [], // TODO
+    filename: httpOperationPart?.filename
+      ? diagnostics.pipe(getSdkModelPropertyType(context, httpOperationPart.filename, operation))
+      : undefined,
+    contentType: httpOperationPart?.body.contentTypeProperty
+      ? diagnostics.pipe(
+          getSdkModelPropertyType(context, httpOperationPart.body.contentTypeProperty, operation)
+        )
+      : undefined,
+  };
+  base.isMultipartFileInput = multiPartOption.isFilePart;
+  base.multipartOptions = operationIsMultipart ? multiPartOption : undefined;
+  if (operationIsMultipart && base.type.kind === "array") {
+    if (httpOperationPart === undefined) {
+      // for "images: bytes[]", return type shall be "bytes" instead of "bytes[]"" and "multipartOptions.multi" shall be true
+      if (base.type.valueType.kind === "bytes") {
+        base.type = base.type.valueType;
+      }
```

Comment:
I think typespec dev asks this behavior for all types.

Reply:
Updated.